pages/api/roles/world/index.ts

```diff
@@ -1,5 +1,105 @@
+import { PrismaClient, Prisma, RoleType } from '@prisma/client';
 import type { NextApiRequest, NextApiResponse } from 'next';
+import { ApiResponse } from '../../../../types/apiResponse';
+import { WorldRoleListData, WorldRoleQueryParams } from '../../../../types/role';
+import { getNonEmptyArrayOrUndefined, combineDefinedArrays, buildFrequencyMap } from '../../../../utils/arrays';
+import { withAuthUser } from '../../../../utils/auth/jwtHelpers';
+import {
+  makeCompanyNameFilters,
+  makeRoleTitleFilters,
+  makeRoleYearFilters,
+  makeRoleTypeFilters,
+} from '../../../../utils/filters/filterHelpers';
+import {
+  convertQueryParamToStringArray,
+  createJsonResponse,
+  HttpMethod,
+  HttpStatus,
+  rejectHttpMethod,
+} from '../../../../utils/http/httpHelpers';
+import { withPrismaErrorHandling } from '../../../../utils/prisma/prismaHelpers';
+import { splitByWhitespaces, splitByCommaRemovingWhitespacesAround } from '../../../../utils/strings/formatters';
 
-export default function handler(req: NextApiRequest, res: NextApiResponse) {
-  res.status(200).end(`Request sent to ${req.url}\n`);
+const prisma = new PrismaClient();
+
+async function handler(userId: string, req: NextApiRequest, res: NextApiResponse) {
+  const method = req.method;
+  switch (method) {
+    case HttpMethod.GET:
+      return handleGet(userId, req, res);
+    default:
+      return rejectHttpMethod(res, method);
+  }
+}
+
+async function handleGet(userId: string, req: NextApiRequest, res: NextApiResponse<ApiResponse<WorldRoleListData[]>>) {
+  const { searchWords, roleTypeWords } = parseGetQueryParams(req);
+
+  const companyNameFilters = makeCompanyNameFilters(searchWords);
+  const roleTitleFilters = makeRoleTitleFilters(searchWords);
+  const roleYearFilters = makeRoleYearFilters(searchWords);
+  const roleTypeFilters = makeRoleTypeFilters(roleTypeWords);
+  const companyOrFilters = companyNameFilters?.map((filter) => ({ company: filter }));
+  const roleTitleOrCompanyFilters = getNonEmptyArrayOrUndefined<Prisma.RoleWhereInput>(
+    combineDefinedArrays<Prisma.RoleWhereInput>([roleTitleFilters, companyOrFilters]),
+  );
+
+  const roles = await prisma.role.findMany({
+    where: {
+      AND: [{ OR: roleTypeFilters }, { OR: roleTitleOrCompanyFilters }, { OR: roleYearFilters }],
+      isVerified: true,
+    },
+    select: {
+      id: true,
+      title: true,
+      type: true,
+      year: true,
+      company: {
+        select: {
+          id: true,
+          name: true,
+          companyUrl: true,
+        },
+      },
+      applications: {
```

Comment:
Currently with our database schema there is no way to tell whether a user is anonymous, although we could support it rather easily.

But either way, I think we don't need to check that since if we filter maybe the data would be lacking. Is your concern about spammy data?